enso/utils.py

```diff
@@ -4,8 +4,46 @@
 from time import strptime
 import pandas as pd
 import numpy as np
-from sklearn.model_selection import StratifiedKFold, StratifiedShuffleSplit
+import random
+from sklearn.model_selection import StratifiedKFold, StratifiedShuffleSplit, train_test_split
 from enso.config import RESULTS_DIRECTORY, FEATURES_DIRECTORY
+from collections import defaultdict
+
+class SafeStratifiedShuffleSplit(StratifiedShuffleSplit):
```

Comment:
This class exists just to make sure that at least one example from each class is put in each split.  StratifiedShuffleSplit doesn't guarantee that by default and it's causing issues at low data volumes